src/components/DateRangePickerInputController.jsx

```diff
@@ -130,14 +133,16 @@ export default class DateRangePickerInputController extends React.Component {
     const {
       startDate,
       isOutsideRange,
+      minimumNights,
       keepOpenOnDateSelect,
       onDatesChange,
     } = this.props;
 
     const endDate = toMomentObject(endDateString, this.getDisplayFormat());
 
     const isEndDateValid = endDate && !isOutsideRange(endDate) &&
-      !isInclusivelyBeforeDay(endDate, startDate);
+      (!isInclusivelyBeforeDay(endDate, startDate) ||
+      (minimumNights === 0 && isSameDay(endDate, startDate)));
```

Comment:
I think this is worth the clone in this scenario. Let's change it to be `isInclusivelyAfterDay(endDate, startDate + minimumNights)`.